lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -551,6 +555,8 @@ data SeqState (n :: NetworkDiscriminant) k = SeqState
         -- (cf: 'PendingIxs')
     , rewardAccountKey :: k 'AddressK XPub
         -- ^ Reward account public key associated with this wallet
+    , derivationPrefix :: DerivationPrefix
+        -- ^ Derivation path prefix from a root key up to the internal account
```

Comment:
There are two major changes in this PR (which then ripples to many other files). One of them is here. `DerivationPrefix` is defined below and is simply a 3-tuple of the first three derivation indexes that leads to the account associated with that state. 

This is necessary because we use seq-state for both Shelley and Icarus which have different `purpose` levels. Also, this will build a nice foundation for the up-coming multi-account support by having the actual account index within the state itself too!